src/Charcoal/CookieConsent/HasCookieConsentTrait.php

```diff
@@ -64,11 +64,15 @@ public function getHasCookieConsent(): bool
     }
 
     /**
-     * @param Container|CookieConsentManager $cookieConsent
+     * @param ContainerInterface|CookieConsentManager $cookieConsent
      * @return void
      */
     public function setCookieConsent($cookieConsent)
     {
-        $this->cookieConsent = $cookieConsent instanceof Container ? $cookieConsent['cookie-consent'] : $cookieConsent;
+        $this->cookieConsent = $cookieConsent;
+
+        if ($cookieConsent instanceof ContainerInterface) {
+            $this->cookieConsent = $cookieConsent['cookie-consent'];
+        }
```

Comment:
This method's logic was bad to begin with.

This method should only accept the `CookieConsentManager`.

Makes no sense to allow the service container.

```php
public function setCookieConsent(CookieConsentManager $cookieConsent): void
{
    $this->cookieConsent = $cookieConsent;
}
```

Furthermore, the property declaration should include and default to `null`. A default value on a property should only be omitted if the value is to be set by the constructor. We can't guarantee this property will be set by the constructor because the constructor might not receive the service container (`__construct(?array $data = null)`).

```diff
- protected CookieConsentManager $cookieConsent;
+ protected ?CookieConsentManager $cookieConsent = null;
```